Fix bearer plugin OCI cloud registries issues

carabasdaniel opened the pull request:
This change should fix https://github.com/open-policy-agent/opa/issues/7092, https://github.com/open-policy-agent/opa/issues/7189 and https://github.com/open-policy-agent/opa/issues/6590

I've tested this with AWS, but I would appreciate if someone can please test this with ACR

ashutosh-narkar commented:
Thanks @carabasdaniel! It would be great if we had a test to exercise this if possible.

ashutosh-narkar commented:
Can you please squash your commits. Are you also able to test this with ACR?

carabasdaniel commented:
> Can you please squash your commits. Are you also able to test this with ACR?

Squashed and rebased on latest main. I'm having some trouble with access to the azure portal at the moment and that is why I requested for someone to test the PR. 

I'll try to test later today if I can get access to ACR

carabasdaniel commented:
Hi @ashutosh-narkar,

I finally managed to test this fix with ACR and it seems to work as expected with a standard private container registry. OPA is able to download the policy image and load the bundle as expected.

ashutosh-narkar commented on v1/plugins/rest/auth.go:
```diff
@@ -153,6 +157,9 @@ func (ap *bearerAuthPlugin) NewClient(c Config) (*http.Client, error) {
 func (ap *bearerAuthPlugin) Prepare(req *http.Request) error {
 	token := ap.Token
+	if ap.logger == nil {
+		ap.logger = logging.Get()
```
I think the logger should be provided. This `Get` call is deprecated. Just noting this if we have to remove it later.
